devices/YarpRobotLoggerDevice/tests/YarpRobotLoggerDeviceTest.cpp

```diff
@@ -0,0 +1,224 @@
+}
+
+std::string toString(const matioCpp::VariableType& type)
+{
+    switch(type)
+    {
+        case matioCpp::VariableType::Element:
+            return "Element";
+        case matioCpp::VariableType::Vector:
+            return "Vector";
+        case matioCpp::VariableType::MultiDimensionalArray:
+            return "MultiDimensionalArray";
+        case matioCpp::VariableType::Struct:
+            return "Struct";
+        case matioCpp::VariableType::CellArray:
+            return "CellArray";
+        case matioCpp::VariableType::StructArray:
+            return "StructArray";
+        case matioCpp::VariableType::Unsupported:
+            return "Unsupported";
+        default:
+            return "Unknown VariableType";
+    }
+}
+
+TEST_CASE("Launch simple logger")
+{
+    // This test launched a dummy logger and verify that it effectively saves some data
+    // Locate location of .exe file of ipit-hipac-codesys-embobj-bridge, done in getDirectoryOfExecutable()
+
+    // As a preliminary step, we rename any leftover .md and .mat files contained in the current working directory
+    // to .md.bak and .mat.bak, to ensure that they are not mistakenly files generated by this tests
+    std::filesystem::path currentPath = std::filesystem::current_path();
+    for (const auto& entry : std::filesystem::directory_iterator(currentPath)) {
+        if (entry.is_regular_file()) {
+            std::filesystem::path filePath = entry.path();
+            std::string extension = filePath.extension().string();
+
+            if (extension == ".mat" || extension == ".md") {
+                std::filesystem::path newFilePath = filePath;
+                newFilePath += ".bak";
+                std::filesystem::rename(filePath, newFilePath);
+            }
+        }
+    }
+
+    // We will be opening yarp ports in the tests, but we do not want to run the yarpserver, so let's set YARP
+    // in local mode (everything will work fine as long as all the ports are opened in the same process)
+    yarp::os::Network network;
+    yarp::os::NetworkBase::setLocalMode(true);
+
+    // To make sure that YarpRobotLoggerDevice is found from the build directory, we add CMAKE_BINARY_DIR
+    // to YARP_DATA_DIRS
+    #ifdef _WIN32
+        std::string envVarListSeparator = ";";
+        #define blf_putenv _putenv
+    #else
+        std::string envVarListSeparator = ":";
+        #define blf_putenv putenv
+    #endif
+    std::string original_yarp_data_dirs = getenv("YARP_DATA_DIRS");
+    std::string putenvArg = "YARP_DATA_DIRS=" + original_yarp_data_dirs + envVarListSeparator + CMAKE_BINARY_DIR + "/share/yarp";
+    blf_putenv(const_cast<char*>(putenvArg.c_str()));
+
+    // We hardcode the yarprobotinterface configuration file is in the source directory and passed via
+    // target_compile_definitions
+    std::filesystem::path pathToXmlConfigurationFile = std::filesystem::path(CMAKE_CURRENT_SOURCE_DIR) / std::filesystem::path("launch-yarp-robot-logger.xml");
```

Comment:
I did not know that CMAKE variables can be used in this way

Reply:
In general they can't, you need to pass them down to the compiler via target_compile_definitions as done in CMake.